src/Nelmio/Alice/Instances/Populator/Methods/ArrayAdd.php

```diff
@@ -62,16 +69,10 @@ private function findAdderMethod($object, $property)
             return $method;
         }
 
-        if (class_exists('Symfony\Component\PropertyAccess\StringUtil') && method_exists('Symfony\Component\PropertyAccess\StringUtil', 'singularify')) {
-            foreach ((array) StringUtil::singularify($property) as $singularForm) {
-                if (method_exists($object, $method = 'add'.$singularForm)) {
-                    return $method;
-                }
-            }
-        } elseif (class_exists('Symfony\Component\Form\Util\FormUtil') && method_exists('Symfony\Component\Form\Util\FormUtil', 'singularify')) {
-            foreach ((array) FormUtil::singularify($property) as $singularForm) {
-                if (method_exists($object, $method = 'add'.$singularForm)) {
-                    return $method;
+        if ($singulars = $this->singularize($property)) {
+            foreach ((array) $singulars as $singularForm) {
```

Comment:
the `(array)` is to take care of the case where `$this->singularize($property)` returns `null`? If so I would rather just make sure we return an array in `::singularize()`